src/colmap/controllers/image_reader.cc

```diff
@@ -146,6 +146,9 @@ ImageReader::Status ImageReader::Next(Camera* camera,
   if (mask && !options_.mask_path.empty()) {
     const std::string mask_path =
         JoinPaths(options_.mask_path, image->Name() + ".png");
+    if (!ExistsFile(mask_path)) {
+      LOG(ERROR) << StringPrintf("Mask Path %s does not exist!", mask_path.c_str());
+    }
     if (ExistsFile(mask_path) && !mask->Read(mask_path, false)) {
       return Status::MASK_ERROR;
```

Comment:
MASK_ERROR already triggers a log message? Is it really necessary to distinguish between file not exists and file is invalid here?

Reply:
See src/colmap/controllers/feature_extraction.cc:265

Reply:
Yes, Read() doesn't tell you what the issue is. It simply says it is bad. Now you have a clear error in the log stating what is wrong in the case of a missing file.

Reply:
OK, fair enough. Could you apply a consistent logic in src/colmap/controllers/feature_extraction.cc:335 then?

Reply:
Check!